mlsdbtool/mcc_update.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+# This will take the mcc data from the file "data/MCC" in the repository
+# tzdata-timed ( https://github.com/sailfishos/tzdata-timed ) and use it
+# to update the code in this repository.
+
+declare -i mcc
+HEADER="../common/mccmapping.h"
+SCRIPT="./mccmapping"
+mcc_file=$1
+if [ -z "$mcc_file" ] ; then
+    echo "Usage: $0 [MCC file]" >&2
+    exit 1
+fi
+if [ ! -e $HEADER ] ; then
+    echo "ERROR: Unable to locate header $HEADER" >&2
+    exit 1
+fi
+if [ ! -e $SCRIPT ] ; then
+    echo "ERROR: Unable to locate script file $SCRIPT" >&2
+    exit 1
+fi
+
+mccs=`cat $mcc_file | grep -v ^# | cut -d ' ' -f1 | sort -nu | tr '\n' ',' | sed 's/^,//;s/,$/,1024/'`
+count=`echo -n $mccs | tr -dc ',' | wc -c`
+
+if [ $count -gt 255 ] ; then
+    echo -e "We have a problem. The functionality of mlsdbtools assumes there are fewer than\n256 unique mccs. The file provided gives us $count. Some manual intervention is needed." >&2
+    exit 1
+fi
+
+echo "Updating $HEADER"
+sed -i "s/=.*/= \{$mccs\}\;/;s/... unique/$count unique/" $HEADER
+
+echo "Updating $SCRIPT"
+last_mcc=0
+first=0
+echo "# MCC -> country code mapping from tzdata-timed repository data/MCC" > $SCRIPT
+for element in `cat $mcc_file | grep -v ^# | cut -d ' ' -f1,3 | sort -u` ; do
+    mcc=$element
+    if [ $mcc -gt 0 ] ; then
```

Comment:
Shellcheck isn't bad for getting a quick sanity check on a script, but it seems quite rigid, and overly anal about double quotes... Often if something would fail because of a lack of quotes, you already have a bigger problem in that the the data you're processing is garbage.

E.g. in this case mcc has been declared to be an integer on line 7, so even if you assigned it to be "green eggs and spam" it would become 0 instead, because the assignment isn't a valid integer. (This is in fact what's being tested here.)

Reply:
Yep. It sometimes gives quite a lot of suggestions. I usually just focus on them more closely, for me it has worked nicely in the past.

But yeah, with expected int comparisons it might not be correct, at least not always.